### PR TITLE
test.py: cluster manager client improve and fix API requests

### PR DESCRIPTION
**Pull request by alecco**
Fix handling of requests by using `async with` for requests.
Fix matching response text across client and server.
Improve manager client handling of request errors.

### CODE REVIEW

**Comment by alecco** [comment]
This pull request is based on top of pending #11364

File: test/pylib/manager_client.py [hunk 1]
```diff
@@ -113,34 +113,34 @@ async def mark_dirty(self) -> None:
         await self._request("/cluster/mark-dirty")
 
     async def server_stop(self, server_id: str) -> bool:
-        """Stop specified node"""
-        ret = await self._request(f"/cluster/node/{server_id}/stop")
+        """Stop specified server"""
```
Comment (nyh) [inline test/pylib/manager_client.py:128]: I am getting a sense of deja-vu. Didn't I already commit this patch? If I did, please rebase this PR.

**Review by nyh — APPROVED** [review]
I'll be happy to commit this, but:
1. Please rebase to get rid of commits that are already in master (I think I recognized the first two...)
2. Please clarify the commit message. It has phrases like "Fix handling of requests by using async with." which explains what was the fix, but not what was the problem that needed fixing...

**Comment by nyh** [comment]
> This pull request is based on top of pending #11364.

This is the wrong link (it links to *this* PR).

**Comment by alecco** [comment]
> > This pull request is based on top of pending #11364.
> 
> This is the wrong link (it links to _this_ PR).

Fixed.
Proper PR base is https://github.com/scylladb/scylladb/pull/11329 it was approved 3 days ago but I don't see it in `master`.
It has 2 commits.
This PR has only 1 commit.

**Comment by nyh** [comment]
Looks like I never committed https://github.com/scylladb/scylladb/pull/11329 despite approving it? I'll take a look at it now.

**Comment by nyh** [comment]
@alecco again we have the
```
>       await self.cql.run_async(f"ALTER TABLE {self.full_name} ADD {column.name} {column.ctype.name}")
E       cassandra.OperationTimedOut: errors={'Connection defunct by heartbeat': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.44.133.9:9042
```

Could it be that you didn't re-create the driver when you should have, or something?

Going just by the error message, maybe changing idle_heartbeat_interval or idle_heartbeat_timeout is needed (see https://docs.datastax.com/en/developer/python-driver/3.24/api/cassandra/cluster/)? I don't know if it needs to be increased or decrease to fix the problem ;-)

**Comment by alecco** [comment]
> @alecco again we have the
> 
> ```
> >       await self.cql.run_async(f"ALTER TABLE {self.full_name} ADD {column.name} {column.ctype.name}")
> E       cassandra.OperationTimedOut: errors={'Connection defunct by heartbeat': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.44.133.9:9042
> ```
> 
> Could it be that you didn't re-create the driver when you should have, or something?
> 
> Going just by the error message, maybe changing idle_heartbeat_interval or idle_heartbeat_timeout is needed (see https://docs.datastax.com/en/developer/python-driver/3.24/api/cassandra/cluster/)? I don't know if it needs to be increased or decrease to fix the problem ;-)

@nyh  Apologies for not updating the pr(s). I was updating `#raft`.

I discussed with Israel and, for _this exception_, we could probably manage by setting a custom `RetryPolicy` and have it return `RETRY_NEXT_HOST` on `.on_write_timeout()` instead of `RETHROW`. The [default RetryPolicy does just plain retry and rethrow for read and write timeouts](https://github.com/scylladb/python-driver/blob/master/cassandra/policies.py#L783).

dtest has a [custom RetryPolicy](https://github.com/scylladb/scylla-dtest/blob/next/dtest_class.py#L22).

I'll send an update later today.

**Comment by nyh** [comment]
Interesting, I didn't know that the default retry policy didn't do the obvious thing of trying a different host.
Please name the new retry policy something more descriptive than dtest's FlakyRetryPolicy (?), and add a comment why you added it :-)

**Comment by alecco** [comment]
Changes in this commit are now conflicting with Kamil's prs, closing and will create a new commit on top of the other prs...

**Pull request closed without merging**